plugins/omdtls/omdtls.c

```diff
@@ -0,0 +1,885 @@
+		} else if(!strcmp(actpblk.descr[i].name, "statsname")) {
+			pData->statsName = (uchar*)es_str2cstr(pvals[i].val.d.estr, NULL);
+		} else if(!strcmp(actpblk.descr[i].name, "tls.authmode")) {
+			char* pszAuthMode = es_str2cstr(pvals[i].val.d.estr, NULL);
+			if(!strcasecmp(pszAuthMode, "fingerprint"))
+				pData->authMode = OSSL_AUTH_CERTFINGERPRINT;
+			else if(!strcasecmp(pszAuthMode, "name"))
+				pData->authMode = OSSL_AUTH_CERTNAME;
+			else if(!strcasecmp(pszAuthMode, "certvalid"))
+				pData->authMode = OSSL_AUTH_CERTVALID;
+			else
+				pData->authMode = OSSL_AUTH_CERTANON;
+			free(pszAuthMode);
+		} else if(!strcmp(actpblk.descr[i].name, "tls.cacert")) {
+			pData->pNetOssl->pszCAFile = (uchar*)es_str2cstr(pvals[i].val.d.estr, NULL);
+			fp = fopen((const char*)pData->pNetOssl->pszCAFile, "r");
+			if(fp == NULL) {
+				char errStr[1024];
+				rs_strerror_r(errno, errStr, sizeof(errStr));
+				LogError(0, RS_RET_NO_FILE_ACCESS,
+				"error: certificate file %s couldn't be accessed: %s\n",
+				pData->pNetOssl->pszCAFile, errStr);
+			} else {
+				fclose(fp);
+			}
+		} else if(!strcmp(actpblk.descr[i].name, "tls.mycert")) {
+			pData->pNetOssl->pszCertFile = (uchar*)es_str2cstr(pvals[i].val.d.estr, NULL);
+			fp = fopen((const char*)pData->pNetOssl->pszCertFile, "r");
+			if(fp == NULL) {
+				char errStr[1024];
+				rs_strerror_r(errno, errStr, sizeof(errStr));
+				LogError(0, RS_RET_NO_FILE_ACCESS,
+				"error: certificate file %s couldn't be accessed: %s\n",
+				pData->pNetOssl->pszCertFile, errStr);
+			} else {
+				fclose(fp);
+			}
+		} else if(!strcmp(actpblk.descr[i].name, "tls.myprivkey")) {
+			pData->pNetOssl->pszKeyFile = (uchar*)es_str2cstr(pvals[i].val.d.estr, NULL);
+			fp = fopen((const char*)pData->pNetOssl->pszKeyFile, "r");
+			if(fp == NULL) {
+				char errStr[1024];
+				rs_strerror_r(errno, errStr, sizeof(errStr));
+				LogError(0, RS_RET_NO_FILE_ACCESS,
+				"error: certificate file %s couldn't be accessed: %s\n",
+				pData->pNetOssl->pszKeyFile, errStr);
+			} else {
+				fclose(fp);
+			}
+		} else if(!strcmp(actpblk.descr[i].name, "tls.tlscfgcmd")) {
+			pData->tlscfgcmd = (uchar*)es_str2cstr(pvals[i].val.d.estr, NULL);
+		} else {
+			LogError(0, RS_RET_INTERNAL_ERROR,
+				"omdtls: program error, non-handled param '%s'\n", actpblk.descr[i].name);
+		}
+	}
+
+	/* check if no port is set. If so, we use DEFAULT of 433 */
+	if(pData->port == NULL) {
+		CHKmalloc(pData->port = (uchar *)strdup("443"));
+	}
+
+	iNumTpls = 1;
+	CODE_STD_STRING_REQUESTnewActInst(iNumTpls);
+
+	CHKiRet(OMSRsetEntry(*ppOMSR, 0, (uchar*)strdup((pData->tplName == NULL) ?
+						"RSYSLOG_FileFormat" : (char*)pData->tplName),
+						OMSR_NO_RQD_TPL_OPTS));
+
+	if (pData->statsName) {
+		CHKiRet(statsobj.Construct(&pData->stats));
+		CHKiRet(statsobj.SetName(pData->stats, (uchar *)pData->statsName));
+		CHKiRet(statsobj.SetOrigin(pData->stats, (uchar *)"omdtls"));
+
+		/* Track following stats */
```

Comment:
It looks like these counters will never be incremented. That needs to be added.

Reply:
Indeed, needs to be added.